scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/SchedulerExamplesUtils.java

```diff
@@ -0,0 +1,148 @@
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.scheduler.examples;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+import org.apache.log4j.Logger;
+
+import com.amazonaws.AmazonClientException;
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.services.s3.transfer.Transfer;
+import com.amazonaws.services.s3.transfer.TransferProgress;
+
+
+/**
+ * SelectionUtils provides static methods for scheduler examples.
+ *
+ * @author ActiveEon Team
+ * @since 17/05/2018
+ */
+public class SchedulerExamplesUtils {
+
+    private static final Logger logger = Logger.getLogger(SchedulerExamplesUtils.class);
+
+    private SchedulerExamplesUtils() {
+
+    }
+
+    /**
+     * waits for the transfer to complete, catching any exceptions that occur.
+      * @param xfer
+     */
+    protected static void waitForCompletion(Transfer xfer) {
+        try {
+            xfer.waitForCompletion();
+        } catch (AmazonServiceException e) {
+            logger.error("Amazon service error: " + e.getMessage());
+            System.exit(1);
+        } catch (AmazonClientException e) {
+            logger.error("Amazon client error: " + e.getMessage());
+            System.exit(1);
+        } catch (InterruptedException e) {
+            logger.error("Transfer interrupted: " + e.getMessage());
+            Thread.currentThread().interrupt();
+            System.exit(1);
+        }
+    }
+
+    /**
+     * Prints progress while waiting for the transfer to finish.
+     * @param xfer
+     */
+    protected static void showTransferProgress(Transfer xfer) {
+        // print the transfer's human-readable description
+        logger.info(xfer.getDescription());
+        // print an empty progress bar...
+        printProgressBar(0.0);
+        // update the progress bar while the xfer is ongoing.
+        do {
+            try {
+                Thread.sleep(100);
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                return;
+            }
+            // Note: so_far and total aren't used, they're just for
+            // documentation purposes.
+            TransferProgress progress = xfer.getProgress();
+            double pct = progress.getPercentTransferred();
+            eraseProgressBar();
+            printProgressBar(pct);
+        } while (!xfer.isDone());
+        // print the final state of the transfer.
+        Transfer.TransferState xferState = xfer.getState();
+        logger.info(": " + xferState);
+    }
+
+    /**
+     * prints a simple text progressbar: [#####     ]
+     * @param pct
+     */
+    protected static void printProgressBar(double pct) {
+        // if bar_size changes, then change erase_bar (in eraseProgressBar) to
+        // match.
+        final int bar_size = 40;
+        final String empty_bar = "                                        ";
```

Comment:
You can use StringBuilder here, it will perform better in terms of space complexity

Reply:
There only one assignment to logMsg. Then, StringBuilder is not very pertinent here.